### PR TITLE
Restructure apps and scripts for docker image builds

### PR DESCRIPTION
**Pull request by DaoWen**
## Changes proposed in this PR

- Reorganized the test-apps directory to make it more flat
- Moved the kubernetes fileserver docker image files into test-apps
- Refactored build-docker-images.sh builds all of our docker images
- Always use "latest" tag for docker images (which we expect local)
- Update integration scrips for smoother experience, both when run locally and when run in Travis.

## Why are we making these changes?

We want to ensure we're always running the latest version of our code/configuration in our test containers when running in Travis, and we want to do that by always running a locally-built "latest" container image.

### CODE REVIEW

**Comment by shamsimam** [comment]
We spoke offline, awaiting directory layout changes.

File: token-syncer/bin/ci/run-integration-tests.sh [hunk 1]
```diff
@@ -16,7 +16,7 @@ TEST_SELECTOR=${2:-integration}
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 PROJECT_DIR=${DIR}/../../../
 WAITER_DIR=${PROJECT_DIR}/waiter
-KITCHEN_DIR=${PROJECT_DIR}/test-apps/kitchen
+KITCHEN_DIR=${PROJECT_DIR}/containers/test-apps/kitchen
```
Comment (DaoWen) [inline token-syncer/bin/ci/run-integration-tests.sh]: I missed updating this path on my first pass after reorganizing the directory structure, but no tests failed. Do we even need this?

**Comment by DaoWen** [comment]
Renamed the old top-level `test-apps` directory to `containers`, moved the test-apps projects into a new `test-apps` subdirectory, and added some READMEs. Ready for review.

File: containers/test-apps/Dockerfile [hunk 2]
```diff
@@ -3,14 +3,11 @@ FROM twosigma/mesos-agent:1.0.0-1.0.1604
...
-  && apt-get install -y nginx \
-  && apt-get install -y python3
+RUN apt-get update && apt-get install -y gettext-base nginx python3
 
-COPY kitchen/bin/waiter-init /usr/bin/waiter-init
-COPY kitchen/bin/kitchen /opt/kitchen/kitchen
+COPY waiter-init/bin/waiter-init /usr/bin/waiter-init
 COPY nginx/bin/run-nginx-server.sh /opt/nginx/bin/run-nginx-server.sh
 COPY nginx/data/nginx-template.conf /opt/nginx/data/nginx-template.conf
+COPY kitchen/bin/kitchen /opt/kitchen/kitchen
```
Comment (shamsimam) [inline containers/test-apps/Dockerfile]: Nitpick: why was this moved lower down than in the previous version?

File: containers/test-apps/waiter-init/bin/waiter-mesos-init [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (shamsimam) [inline containers/test-apps/waiter-init/bin/waiter-mesos-init]: Can we delete this file?

File: token-syncer/bin/ci/run-integration-tests.sh (hunk 1 above)
Comment (shamsimam) [reply token-syncer/bin/ci/run-integration-tests.sh]: I'm surprised that it worked, the `test-ping-tokens` should use it.

Comment (DaoWen) [reply token-syncer/bin/ci/run-integration-tests.sh]: I checked the logs and the tests were skipped. Mystery solved.

File: containers/test-apps/Dockerfile (hunk 2 above)
Comment (DaoWen) [reply containers/test-apps/Dockerfile]: I think this is an artifact of me splitting this off from my WAITER_SANDBOX PR. I'll sort these lexicographically instead.

File: containers/test-apps/waiter-init/bin/waiter-mesos-init (hunk 3 above)
Comment (DaoWen) [reply containers/test-apps/waiter-init/bin/waiter-mesos-init]: Yes, this was supposed to be added in the WAITER_SANDBOX PR. Deleted.
